### DISCUSSION TITLE
Window interaction based effects

### DISCUSSION THREAD

**Discussion by rocksdanister** [CommunityToolkit/Labs-Windows#468] [Ideas]
### Describe the problem
Creating controls or UI that reacts to user action or surroundings would be fun.

### Describe the solution
Here are examples using ComputeSharp shaders

**Window position change**

https://github.com/CommunityToolkit/Labs-Windows/assets/17554161/ca2406b1-36ff-4a5f-9e12-cc365af1d2fe

**Window activation change**

https://github.com/CommunityToolkit/Labs-Windows/assets/17554161/a7dcab27-d0a4-425e-ab15-f6c007d73c3c

**Window shake detection**

https://github.com/CommunityToolkit/Labs-Windows/assets/17554161/c200e35a-94c9-4bec-bc93-a1d61b5fe612

Similarly effects for window size change etc.. for UI elements.

### Additional info
There will be api differences between frameworks, for example WinUI has [Window position events](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindow.changed?view=windows-app-sdk-1.3) but UWP does not have similar event.

**Comment by rocksdanister** [comment]
Related https://github.com/CommunityToolkit/Labs-Windows/discussions/467

**Comment by rocksdanister** [comment]
I made an app that has some of these features:
https://github.com/rocksdanister/weather

**Reply by michael-hawker** [reply]
Thanks for sharing @rocksdanister, your apps are awesome! 😊

Did you have ideas for how these APIs would be exposed as a control/helper? Or are you looking for help from someone else to extract what you've built within your app and proposed here?
